xmldiff/diff.py

```diff
@@ -162,23 +162,31 @@ def node_ratio(self, left, right):
             # One is a comment the other is not:
             return 0
 
+        match = None
         for attr in self.uniqueattrs:
-            if not isinstance(attr, str):
-                # If it's actually a sequence of (tag, attr), the tags must
-                # match first.
-                tag, attr = attr
-                if tag != left.tag or tag != right.tag:
-                    continue
-            if attr in left.attrib or attr in right.attrib:
-                # One of the nodes have a unique attribute, we check only that.
+            if isinstance(attr, dict):
+                for tag, attr_list in attr.items():
+                    if tag != left.tag or tag != right.tag:
+                        continue
+                    matches = 0
+                    attr_list_consolidated = \
+                        [attr for attr in attr_list
+                         if attr in left.attrib or attr in right.attrib]
+                    for inner_attr in attr_list_consolidated:
+                        matches += int(left.attrib.get(inner_attr)
+                                       == right.attrib.get(inner_attr))
+                    match = int(matches/len(attr_list) == 1)
+            elif attr in left.attrib or attr in right.attrib:
+                # One of the nodes have a unique attribute,
+                # we check only that.
                 # If only one node has it, it means they are not the same.
-                return int(left.attrib.get(attr) == right.attrib.get(attr))
+                match = int(left.attrib.get(attr) == right.attrib.get(attr))
+        if match is None:
```

Comment:
No, this is wrong. If one side has a unique attribute, it's either a full match or not a match. Nothing else needs to be checked. You can loop through the unique attributes, and if one side has one, and the other don't, then you can exit the whole function with a 0.